vmware-tanzu/tanzu-framework
# Allow management-cluster creation using ClusterClass when `package-based-lcm` flag is enabled

### What this PR does / why we need it

- This PR fixes issues for creating management-cluster using `package-based-lcm` approach
- Fixes includes:
    - Set featuregate for addons ClusterBootstrapController to true
    - Install clusterclass to tkg-system namespace
    - Fix kapp-controller package values
    - Install kapp-controller before installing providers
    - Do not wait for addons deployment if `package-based-lcm` feature flag is enabled
    - Apply overlays based on pblcm flag and Remove unneccesary overlays from yttcc directory
    - Add patch awsclustertemplate for cni-ingress as part of clusterclass
    - Create cluster template after deploying clusterclass with packages
    - Do not wait for packages on mgmt-cluster if package-based-lcm ff is enabled

…

### Which issue(s) this PR fixes


Fixes #2358

### Describe testing done for PR

1. Build all management-components along with management package repository
2. Use the management package repository to create management-cluster
3. Make sure management-cluster gets created successfully using clusterclass



### Release note

```release-note
package-based-lcm: Allow management-cluster creation using ClusterClass
```

### PR Checklist



      


### Additional information

#### Special notes for your reviewer
